sktime/annotation/hmm_learn.py

```diff
@@ -0,0 +1,232 @@
+        Determines if the forward-backward algorithm is implemented with
+        logarithms ("log"), or using scaling ("scaling").  The default is
+        to use logarithms for backwards compatability.
+
+    Attributes
+    ----------
+    n_features : int
+        Dimensionality of the Gaussian emissions.
+    monitor_ : ConvergenceMonitor
+        Monitor object used to check the convergence of EM.
+    startprob_ : array, shape (n_components, )
+        Initial state occupation distribution.
+    transmat_ : array, shape (n_components, n_components)
+        Matrix of transition probabilities between states.
+    means_ : array, shape (n_components, n_features)
+        Mean parameters for each state.
+    covars_ : array
+        Covariance parameters for each state.
+        The shape depends on :attr:`covariance_type`:
+        * (n_components, )                        if "spherical",
+        * (n_components, n_features)              if "diag",
+        * (n_components, n_features, n_features)  if "full",
+        * (n_features, n_features)                if "tied".
+
+    Examples
+    --------
+    >>> from sktime.annotation.hmm_learn import GaussianHMM
+    >>> from sktime.annotation.datagen import piecewise_normal
+    >>> data = piecewise_normal(
+    ...    means=[2, 4, 1], lengths=[10, 35, 40], random_state=7
+    ...    ).reshape((-1, 1))
+    >>> model = GaussianHMM(algorithm='viterbi', n_components=2)
+    >>> model = model.fit(data)
+    >>> labeled_data = model.predict(data)
+    """
+
+    def __init__(
+        self,
+        n_components: int = 1,
+        covariance_type: str = "diag",
+        min_covar: float = 1e-3,
+        startprob_prior: float = 1.0,
+        transmat_prior: float = 1.0,
+        means_prior: float = 0,
+        means_weight: float = 0,
+        covars_prior: float = 1e-2,
+        covars_weight: float = 1,
+        algorithm: str = "viterbi",
+        random_state: float = None,
+        n_iter: int = 10,
+        tol: float = 1e-2,
+        verbose: bool = False,
+        params: str = "stmc",
+        init_params: str = "stmc",
+        implementation: str = "log",
+    ):
+
+        self.n_components = n_components
+        self.covariance_type = covariance_type
+        self.min_covar = min_covar
+        self.startprob_prior = startprob_prior
+        self.transmat_prior = transmat_prior
+        self.means_prior = means_prior
+        self.means_weight = means_weight
+        self.covars_prior = covars_prior
+        self.covars_weight = covars_weight
+        self.algorithm = algorithm
+        self.random_state = random_state
+        self.n_iter = n_iter
+        self.tol = tol
+        self.verbose = verbose
+        self.params = params
+        self.init_params = init_params
+        self.implementation = implementation
+        super(GaussianHMM, self).__init__()
+
+    def _fit(self, X, Y=None):
+        # import inside _fit to avoid hard dependency.
```

Comment:
I'm wondering what type of expected type X is (i.e. pd.DaraFrame/Series or np.Array) - a docstring?

Reply:
Yep, there should be some docstrings here.  Adding them now!